### PR TITLE
feat(auto-detect): add basic auto-detection for cargo-dist support

### PR DESCRIPTION
**Pull request by Gankra**

### CODE REVIEW

File: src/config/project/mod.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ pub struct ProjectConfig {
     pub repository: Option<String>,
     pub version: Option<String>,
     pub license: Option<String>,
+    pub readme_path: Option<Utf8PathBuf>,
```
Comment (ashleygwilliams) [inline src/config/project/mod.rs:18]: can you say more about adding this here? does this connect to axoproject? (it's not clear to me we ever receive this value when deserializing a project manifest so i want to make sure it's connected)

**Review by ashleygwilliams — COMMENTED** [review]
i am not entirely sure if the refactor maintains all the previous logic, as somethings have defaults and others don't. is this still using defaults where available?

File: src/config/project/mod.rs (hunk 1 above)
Comment (Gankra) [reply src/config/project/mod.rs:18]: It's a value that Cargo.toml lets you specify and axoproject already reads. I did a quick pass of "are there values we're not exposing that we already have equivalents of in oranda" and this is the only one that turned up.

Comment (ashleygwilliams) [reply src/config/project/mod.rs:18]: does axoproject do it for js files as well? package.json has a readme key - i am cool with adding this, just want to make sure that it's fully baked :)

**Review by manyinsects — APPROVED** [review]

**Pull request merged**